lib/r10k/environment/git.rb

```diff
@@ -38,8 +40,16 @@ class R10K::Environment::Git < R10K::Environment::WithModules
   # @param options [String] :ref The git reference to use for this environment
   def initialize(name, basedir, dirname, options = {})
     super
-    @remote = options[:remote]
-    @ref    = options[:ref]
+    setopts(options, {
+      # Standard option interface
+      :version => :ref,
+      :source  => :remote,
+      :type    => ::R10K::Util::Setopts::Ignore,
+
+      # Type-specific options
+      :ref     => :self,
+      :remote  => :self,
+    }, raise_on_unhandled: false)
```

Comment:
Because previously, git-based environments would let you pass garbage options. So for backwards compatibility, we still let you pass garbage options, here, for this one object type. 🤦

Reply:
oof. Want to add a comment to that effect? Then ideally we could remove that if we ever release a new major version.

Reply:
Will do